### ISSUE TITLE
[BUG] Missing required key 'FromAddress' in params.EmailChannelRequest

### ISSUE THREAD

**Issue by Ricardo1980** [aws-amplify/amplify-cli#7464]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

v15.1.0

### Amplify CLI Version

4.51.3

### What operating system are you using?

macOS 10.15.7

### Amplify Categories

auth, storage, function, api, analytics, notifications

### Amplify Commands

pull

### Describe the bug

Every time I run "amplify pull" and I select "yes" to the question:
"? Do you plan on modifying this backend? Yes"
then I receive this error:
<img width="1109" alt="Screenshot 2021-06-06 at 13 42 14" src="https://user-images.githubusercontent.com/738681/120925089-c411de00-c6ce-11eb-88f0-8fa13a05a31a.png">

However, if I select NO, I don't receive the error.
I tested this in another computer, and same thing.
I also see that there is another ticket with the same error, but the issue was not fixed:
https://github.com/aws-amplify/amplify-cli/issues/5848
https://github.com/aws-amplify/amplify-cli/issues/6371

### Expected behavior

The full backend is downloaded.

### Reproduction steps

I can only reproduce the error with my project.
If I create a new one, then I don't see the error.
Do you need the id of my project?

### GraphQL schema(s)

NA

### Log output

<details>

```
# Put your logs below this line
2021-06-06T12:51:14.938Z|info : amplify pull core  
2021-06-06T12:53:03.596Z|info : amplify-provider-awscloudformation.system-config-manager.getProfileConfig(["aws_admin_profile"])
2021-06-06T12:53:03.599Z|info : amplify-provider-awscloudformation.system-config-manager.getProfiledAwsConfig.profileConfig([{"region":"eu-west-1"}])
2021-06-06T12:53:03.599Z|info : amplify-provider-awscloudformation.system-config-manager.getProfileCredentials(["aws_admin_profile"])
2021-06-06T12:53:03.612Z|info : amplify-provider-awscloudformation.amplify-service-permission-check.checkAmplifyServiceIAMPermission.amplifyClient.listApps([])
2021-06-06T12:53:03.775Z|info : amplify-provider-awscloudformation.attach-backend.getAmplifyApp.amplifyClient.listApps([{"maxResults":25}])
2021-06-06T12:53:05.218Z|info : amplify-provider-awscloudformation.attach-backend.getBackendEnv.amplifyClient.listBackendEnvironments([{"appId":"[***]n7i7fu"}])
2021-06-06T12:53:05.350Z|info : amplify-provider-awscloudformation.attach-backend.downloadBackend.s3.getObject([{"Key":"[***]rent-[***]ud-[***]d.zip","Bucket":"[***]ify-[***]ish-[***]st-[***]ev-[***]117-[***]ment"}])
2021-06-06T12:53:11.434Z|info : amplify-provider-awscloudformation.system-config-manager.getProfileConfig(["aws_admin_profile"])
2021-06-06T12:53:11.435Z|info : amplify-provider-awscloudformation.system-config-manager.getProfiledAwsConfig.profileConfig([{"region":"eu-west-1"}])
2021-06-06T12:53:11.435Z|info : amplify-provider-awscloudformation.system-config-manager.getProfileCredentials(["aws_admin_profile"])
2021-06-06T12:53:11.437Z|info : amplify-provider-awscloudformation.system-config-manager.getProfileConfig(["aws_admin_profile"])
2021-06-06T12:53:11.438Z|info : amplify-provider-awscloudformation.system-config-manager.getProfiledAwsConfig.profileConfig([{"region":"eu-west-1"}])
2021-06-06T12:53:11.438Z|info : amplify-provider-awscloudformation.system-config-manager.getProfileCredentials(["aws_admin_profile"])
2021-06-06T12:53:11.444Z|info : amplify-provider-awscloudformation.zip-util.downloadZip.s3.getFile([{"Key":"[***]rent-[***]ud-[***]d.zip"},null])
2021-06-06T12:53:11.447Z|info : amplify-provider-awscloudformation.aws-s3.s3.getFile([{"Key":"[***]rent-[***]ud-[***]d.zip","Bucket":"[***]ify-[***]ish-[***]st-[***]ev-[***]117-[***]ment"}])
2021-06-06T12:53:12.159Z|info : amplify-provider-awscloudformation.initialize-env.run.cfn.updateamplifyMetaFileWithStackOutputs([{"StackName":"[***]ify-[***]ish-[***]st-[***]ev-[***]117"}])
2021-06-06T12:53:12.160Z|info : amplify-provider-awscloudformation.aws-cfn.updateamplifyMetaFileWithStackOutputs.cfn.describeStackResources([{"StackName":"[***]ify-[***]ish-[***]st-[***]ev-[***]117"}])
2021-06-06T12:53:12.583Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:eu-[***]st-[***]63:stack/[***]ify-[***]ish-[***]st-[***]ev-[***]117-[***]ntResource-[***]SPRXJ7/[***]b9f0-[***]66-[***]ea-[***]c6-[***]8180c"}])
2021-06-06T12:53:12.585Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:eu-[***]st-[***]63:stack/[***]ify-[***]ish-[***]st-[***]ev-[***]117-[***]testapi-[***]25NNE/[***]ff60-[***]a7-[***]ea-[***]0c-[***]712e0"}])
2021-06-06T12:53:12.586Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:eu-[***]st-[***]63:stack/[***]ify-[***]ish-[***]st-[***]ev-[***]117-[***]Resource-[***]T8F52R/[***]ea80-[***]ad-[***]eb-[***]f7-[***]2ec0a"}])
2021-06-06T12:53:12.587Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:eu-[***]st-[***]63:stack/[***]ify-[***]ish-[***]st-[***]ev-[***]117-[***]tcd6454bd-[***]FAPZR8/[***]fa10-[***]47-[***]ea-[***]0c-[***]323aa"}])
2021-06-06T12:53:12.588Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:eu-[***]st-[***]63:stack/[***]ify-[***]ish-[***]st-[***]ev-[***]117-[***]reateAuthChallenge-[***]CHTEK/[***]7ec0-[***]47-[***]ea-[***]4a-[***]2d5e8"}])
2021-06-06T12:53:12.589Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:eu-[***]st-[***]63:stack/[***]ify-[***]ish-[***]st-[***]ev-[***]117-[***]efineAuthChallenge-[***]Z7MFGP/[***]4d30-[***]47-[***]ea-[***]0c-[***]712e0"}])
2021-06-06T12:53:12.590Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:eu-[***]st-[***]63:stack/[***]ify-[***]ish-[***]st-[***]ev-[***]117-[***]454bdPreSignUp-[***]BGB5S/[***]6e00-[***]47-[***]ea-[***]22-[***]840dc"}])
2021-06-06T12:53:12.591Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:eu-[***]st-[***]63:stack/[***]ify-[***]ish-[***]st-[***]ev-[***]117-[***]yAuthChallengeResponse-[***]BX677W/[***]69d0-[***]47-[***]ea-[***]e3-[***]2f99e"}])
2021-06-06T12:53:12.592Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:eu-[***]st-[***]63:stack/[***]ify-[***]ish-[***]st-[***]ev-[***]117-[***]gesFunction-[***]CR21UE/[***]dcd0-[***]ff-[***]eb-[***]aa-[***]44edb"}])
2021-06-06T12:53:12.593Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:eu-[***]st-[***]63:stack/[***]ify-[***]ish-[***]st-[***]ev-[***]117-[***]rGameResource-[***]OX3AJT/[***]0b10-[***]26-[***]eb-[***]a8-[***]e99ba"}])
2021-06-06T12:53:12.594Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:eu-[***]st-[***]63:stack/[***]ify-[***]ish-[***]st-[***]ev-[***]117-[***]bdaLayer-[***]XZ8EF6/[***]28a0-[***]aa-[***]eb-[***]db-[***]bd58b"}])
2021-06-06T12:53:12.594Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:eu-[***]st-[***]63:stack/[***]ify-[***]ish-[***]st-[***]ev-[***]117-[***]NotificationFunction-[***]GCBHI/[***]8ef0-[***]d9-[***]eb-[***]23-[***]8aec1"}])
2021-06-06T12:53:12.595Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:eu-[***]st-[***]63:stack/[***]ify-[***]ish-[***]st-[***]ev-[***]117-[***]eiptResource-[***]9S9OMQ/[***]4650-[***]9c-[***]eb-[***]da-[***]d48dd"}])
2021-06-06T12:53:12.596Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:eu-[***]st-[***]63:stack/[***]ify-[***]ish-[***]st-[***]ev-[***]117-[***]veAccount-[***]ODX55/[***]9f50-[***]c4-[***]eb-[***]58-[***]235f1"}])
2021-06-06T12:53:12.597Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:eu-[***]st-[***]63:stack/[***]ify-[***]ish-[***]st-[***]ev-[***]117-[***]tionResource-[***]25RSU/[***]0d10-[***]4d-[***]ea-[***]38-[***]6cfe8"}])
2021-06-06T12:53:12.598Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:eu-[***]st-[***]63:stack/[***]ify-[***]ish-[***]st-[***]ev-[***]117-[***]veVisit-[***]4HIKXA/[***]0e80-[***]84-[***]ea-[***]76-[***]dc706"}])
2021-06-06T12:53:12.598Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:eu-[***]st-[***]63:stack/[***]ify-[***]ish-[***]st-[***]ev-[***]117-[***]ageFunction-[***]8O9N3/[***]7960-[***]4b-[***]eb-[***]ba-[***]ad013"}])
2021-06-06T12:53:12.599Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:eu-[***]st-[***]63:stack/[***]ify-[***]ish-[***]st-[***]ev-[***]117-[***]ortResource-[***]7QP2D/[***]3900-[***]fa-[***]eb-[***]15-[***]5464c"}])
2021-06-06T12:53:12.600Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:eu-[***]st-[***]63:stack/[***]ify-[***]ish-[***]st-[***]ev-[***]117-[***]essageResource-[***]RXW61/[***]d070-[***]0d-[***]eb-[***]5f-[***]d5910"}])
2021-06-06T12:53:12.601Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:eu-[***]st-[***]63:stack/[***]ify-[***]ish-[***]st-[***]ev-[***]117-[***]onEmailResource-[***]AO2P0H/[***]b890-[***]91-[***]eb-[***]38-[***]523d0"}])
2021-06-06T12:53:12.602Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:eu-[***]st-[***]63:stack/[***]ify-[***]ish-[***]st-[***]ev-[***]117-[***]kenResource-[***]7NS9Q/[***]07e0-[***]ad-[***]eb-[***]6a-[***]e26ca"}])
2021-06-06T12:53:12.602Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:eu-[***]st-[***]63:stack/[***]ify-[***]ish-[***]st-[***]ev-[***]117-[***]54971a4-[***]YK0MNM/[***]6160-[***]30-[***]ea-[***]5a-[***]bde12"}])
2021-06-06T12:53:12.809Z|info : amplify-provider-awscloudformation.aws-s3.s3.getFile([{"Key":"[***]ify-[***]json","Bucket":"[***]ify-[***]ish-[***]st-[***]ev-[***]117-[***]ment"}])
2021-06-06T12:53:12.942Z|info : amplify-provider-awscloudformation.system-config-manager.getProfileConfig(["aws_admin_profile"])
2021-06-06T12:53:12.942Z|info : amplify-provider-awscloudformation.system-config-manager.getProfiledAwsConfig.profileConfig([{"region":"eu-west-1"}])
2021-06-06T12:53:12.942Z|info : amplify-provider-awscloudformation.system-config-manager.getProfileCredentials(["aws_admin_profile"])
2021-06-06T12:53:13.322Z|info : amplify-provider-awscloudformation.system-config-manager.getProfileConfig(["aws_admin_profile"])


```

</details>

**Comment by jhockett** [comment]
Hi @Ricardo1980, I wasn't able to reproduce this error in a new project like you mentioned. When you run `amplify pull`, do you end up with any project files?

**Comment by Ricardo1980** [comment]
@jhockett 
There are 2 cases, depending on my reply to this question:
Do you plan on modifying this backend? 
**If I select yes:**
I see the crash and also `amplifyconfiguration.json` and `awsconfiguration.json` are downloaded, but not the `amplify` folder, which is wrong because it should be downloaded.
**If I select no:**
Then `amplifyconfiguration.json` and `awsconfiguration.json` are downloaded and no more files, which seems it is expected and right.

Please, I need help with this, if I lose my laptop, then I will lose my project too, given that is impossible to download it anywhere else.

**Comment by akshbhu** [comment]
Hi @Ricardo1980 

Can you provide me the steps or the categories you added when you got this error?

**Comment by Ricardo1980** [comment]
@akshbhu @jhockett
I have something better, a video!
https://www.dropbox.com/s/1v8q8ozeebixtc9/demo.mov?dl=0
You can see that the amplify folder is created while doing all the steps, but removed after the crash. Only the 2 configuration files remain.

Note that I think you won't be able to reproduce it, I guess it is something about my project.
So, in that case, can anyone from AWS check it?

Thanks.

**Comment by edwardfoyle** [comment]
The fix for this has been released in version 5.0.2 of the CLI. Please upgrade to the latest version of the CLI.
